### PR TITLE
RATIS-1766. Add descriptions to metrics entries

### PR DESCRIPTION
**Pull request by SzyWilliam**
## What changes were proposed in this pull request?

Add descriptions to metrics entries

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/RATIS-1766

### CODE REVIEW

**Comment by SzyWilliam** [comment]
@szetszwo @codings-dan I've added descriptions to the existing metrics. Please take a look at its correctness if you have time. Thanks very much in advance!

File: ratis-docs/src/site/markdown/metrics.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+-->
+
+
+# Metrics
+
+## Ratis Server
+
+### StateMachine Metrics
+
+| Application | Category      | Name                | Type  | Description                                                  |
```
Comment (szetszwo) [inline ratis-docs/src/site/markdown/metrics.md:25]: To be consistent with the code, let's rename `Category` to `Component`.

**Review by szetszwo — COMMENTED** [review]
@SzyWilliam , thanks a lot for working on this!  The change looks great.  Just one comment inlined.

**Review by codings-dan — APPROVED** [review]
@SzyWilliam  Thanks for working on this! the change looks good to me, +1

**Comment by SzyWilliam** [comment]
@szetszwo @codings-dan Thanks for the reviews! Made changes on doc, PTAL~

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**